### PR TITLE
Add LayerNormalization to tfjs

### PR DESCRIPTION
**Pull request by Derek-Wds**
This is about issue [#1548](https://github.com/tensorflow/tfjs/issues/1548).

---
This change is [Reviewable](https://reviewable.io/reviews/tensorflow/tfjs-layers/545)

### CODE REVIEW

**Comment by Derek-Wds** [comment]
@caisq Hi, thanks for you reply. I will make some changes soon, but I have a question and I hope you could help me with it:
* Since LayerNormalization is doing the same thing as BatchNormalization but on different dimensions, and it uses the function of batch norm. In the unit test of BatchNormalization, it seems that all the tests are based on `functions` instead of BatchNormalization `class`. I wonder if there have other ways to test the LayerNormalization `class`.

Thanks,
Derek

**Comment by caisq** [comment]
@Derek-Wds It's not true that all tests in `normalization_test.ts` are based on functions and not classes. Please look at the test that use `tfl.layers.batchNormalization()`. It may appear to be a function call, but that function actually is a factory function that calls the constructor 'BatchNormalization' under the hood. So it's testing the class.

https://github.com/tensorflow/tfjs-layers/blob/master/src/layers/normalization_test.ts#L301

**Comment by Derek-Wds** [comment]
@caisq Thanks for your response! I'm looking into that.

**Comment by Derek-Wds** [comment]
@caisq Hi, I have added unit test and change the implementation. Can you help me take a look at the error? I cannot run test on my machine (it says something went wrong with the [karma-server]).

Thanks,
Derek

**Comment by caisq** [comment]
@Derek-Wds I'm curious why you closed this PR. Did you run into any trouble? If so, is there anything I can help with? I think LayerNormlization is an important enough new layer type that it should be added.
